### ISSUE TITLE
wrong row order after initialization

### ISSUE THREAD

**Issue by philly-vanilly** [ceolter/ag-grid-angular#157]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid-angular/blob/master/CONTRIBUTING.md#question
```
I asked this on stackoverflow, see https://stackoverflow.com/questions/44841350/ag-grid-angular-wrong-row-order-after-initialization and since I got no reply I assume this is a bug and I am not doing something wrong, so I am opening this issue. You guys do an amazing job and if there is a mistake on my side, sorry for that!

**Expected behavior**
When I initialize the grid and get hold of the elements in it, they should be returned/iterated in the displayed order, starting with the first one and not the eleventh.

**Minimal reproduction of the problem with instructions**
After initializing aggrid with

```
this.gridOptions = {
  enableSorting: true,
  rowHeight: 50,
  suppressDragLeaveHidesColumns: true,
  enableColResize: true,
  domLayout: 'autoHeight',
  rowSelection: 'single',
};
```
and

```
  <ag-grid-angular
    #agGrid
    [rowData]="articleArray"
    [gridOptions]="gridOptions"
    (cellClicked)="showDetails($event)"
    (gridReady)="onGridReady()"
    (modelUpdated)="onModelUpdated()"
    (gridSizeChanged)="sizeToFit()"
  ></ag-grid-angular>
```

And getting the resulted order of rows in the grid by

```
  updateArticleOrder() {
    const result: string[] = [];
    this.gridOptions.api.forEachNodeAfterFilterAndSort(rowNode => result.push(rowNode.data.id));
    this.articleOrderChanged$.emit(result);
  }
```
I notice that the order is different from the original input used in [rowData]="articleArray". To be precise, 11 items have been shifted to the bottom of the grid and the grid starts at original index 11. Why is that and how can I avoid it? After sorting and updateArticleOrder() being called inside onModelUpdated() I get the right order again, with index 0 displayed when I click the first row.

I notice that I get this problem only when initializing the grid. When my articleArray updates after the grid is already visible, this.gridOptions.api.forEachNodeAfterFilterAndSort gives me the new items in the correct order.

**Please tell us about your environment:**
TLDR: Angular 4, aggrid 10.1.0, NGRX-store in the background
```
  "dependencies": {
    "@angular/animations": "4.0.3",
    "@angular/common": "4.0.3",
    "@angular/compiler": "4.0.3",
    "@angular/core": "4.0.3",
    "@angular/forms": "4.0.3",
    "@angular/http": "4.0.3",
    "@angular/platform-browser": "4.0.3",
    "@angular/platform-browser-dynamic": "4.0.3",
    "@angular/router": "4.0.3",
    "@ngrx/core": "1.2.0",
    "@ngrx/db": "2.0.2",
    "@ngrx/effects": "2.0.3",
    "@ngrx/router-store": "1.2.6",
    "@ngrx/store": "2.2.2",
    "@ngrx/store-devtools": "3.2.4",
    "ag-grid": "10.1.0",
    "ag-grid-angular": "10.1.0",
    "angular2-jwt": "0.2.3",
    "angular2-take-until-destroy": "1.0.3",
    "angular2-toaster": "4.0.0",
    "asn1": "0.2.3",
    "bootstrap": "4.0.0-alpha.6",
    "co": "4.6.0",
    "core-js": "2.4.1",
    "cross-spawn": "5.1.0",
    "font-awesome": "4.7.0",
    "json-stable-stringify": "1.0.1",
    "moment": "2.18.1",
    "ng-inline-svg": "3.2.1",
    "primeng": "4.0.3",
    "reselect": "3.0.0",
    "rxjs": "5.1.0",
    "zone.js": "0.8.4"
  }
```

**Comment by seanlandsman** [comment]
Hi,

forEachNodeAfterFilterAndSort should return the data in display order, not necessarily in the order supplied.

Does your grid have any sorting applied? If not then the order should be in the order the data is supplied

**Comment by philly-vanilly** [comment]
Hi, I was using forEachNodeAfterFilterAndSort exactly to do that - get the data in the displayed order. The bug seems to come from an overflowing buffer and disappeared when I set the buffer to a high value (see stackoverflow).

**Comment by seanlandsman** [comment]
Hi,

Ok, I see - I'll spend a little time trying to reproduce this (probably a bit later in the week) and if its indeed an issue I'll raise an issue for it our side

thanks - will update this thread with any new info

**Comment by rmc-software** [comment]
Hi - this is as a consequence of using domLayout: 'autoHeight'. Auto height is intended for a particular use case only and as there is a workaround using rowBuffer, or even using a setTimeout(), I don't think we'll look to address this issue.

**Comment by aaronjedwards** [comment]
I am also running into this same issue with `domLayout: 'autoHeight'`. This seems like a bug to me as it is inconsistently happening. Is this not going to be looked into further?
